## General Parameters
name: ekf1
num_threads: 1
record_video: false

## Initial State
x0: [0, 0, -2,    # POS
     0, 0, 0,         # VEL
     1, 0, 0, 0,      # ATT
     0.0, 0.0, 0.0, # B_A
     0.0, 0.0, 0.0,  # B_G
     0.1]           # MU

## Initial Uncertainty
P0: [1e-3, 1e-3, 1e-3,
     1e-3, 1e-3, 1e-3,
     1e-3, 1e-3, 1e-3,
     2e-1, 2e-1, 2e-1,
     1e-1, 1e-1, 1e-1,
     1e-6]
P0_feat: [0.01, 0.01, 0.3]

## Additive Process Noise
Qx: [0.0, 0.0, 0.0,
     0.0, 0.0, 0.0,
     0.0, 0.0, 0.0,
     0.0, 0.0, 0.0,
     0.0, 0.0, 0.0,
     0.0]
Qx_feat: [0.0, 0.0, 0.0]

# Input Process Noise
Qu: [1e0, 1e0, 1e0,      # u_acc
     4e-4, 4e-4, 4e-4]  # u_gyro

## Partial Update Gain
lambda: [1.0, 1.0, 1.0,     # POS
         1.0, 1.0, 1.0,     # VEL
         1.0, 1.0, 1.0,     # ATT
         0.1, 0.1, 0.1,     # B_A
         0.1, 0.1, 0.1,     # B_G
         0.01]               # MU
lambda_feat: [1.0, 1.0, 0.4]

## Sensor Noise
depth_R: 0.5
feat_R: [10.0, 10.0]
acc_R_drag: [1.0, 1.0]
acc_R_grav: [1.0, 1.0, 1.0]
pos_R: [0.1, 0.1, 0.1]
vel_R: [0.1, 0.1, 0.1]
att_R: [0.01, 0.01, 0.01]
alt_R: 0.1

## Low-Pass Filtering on IMU and Truth
imu_LPF: 0.0
truth_LPF: 0.0

## Measurement Model Flags
use_truth: false
use_depth: false
use_features: true
use_acc: true
use_imu_att: false
use_alt: false

## Feature Flags
use_drag_term: true
use_partial_update: true
use_keyframe_reset: true

## Image feature tracking Parameters
min_depth: 1.5
keyframe_overlap_threshold: 0.8
num_features: 20
feature_radius: 45

## Camera parameters
image_size: [640, 480]
cam_center: [315.83184814453125, 242.1165771484375]
focal_len: [611.1864013671875, 611.5557861328125]
distortion: [0, 0, 0, 0]
invert_image: false

q_b_c: [0.9974335273839892, 0.019768487288642146, -0.03564306227555538, 0.05886541830542371]
p_b_c: [0.17363129, -0.02205945,  0.05490228]

feature_mask: ../params/mask_default.png

## IMU and Mocap rotations
q_b_u: [0.993760669165504, 0.0497294816014604, 0.0997086508721388, 0.00498959122946198]
q_I_m: [0.9747941070689433, 0.0, 0.22310636213174545, 0.0]
